### ISSUE TITLE
Issue with crypting api key

### ISSUE THREAD

**Issue by catalindumitrache76** [UNIZAR-30246-WebEngineering/lab2-big-ws#27]
Hi.

I'm stuck at step number 4, where it sais that we must discover how to commit our code to GitHub and allow Travis to test it when there is sensitive data involved ... 

For this step I had to download Ruby on my PC, and afterwards I used the following command:
"_travis encrypt <<VAR_NAME>>=API KEY_". 

That command generates this output:
"_Please add the following to your .travis.yml file:
 secure:<<ENCRYPTED_KEY>>_

Pro Tip: You can add it automatically by running with --add."

Well .. I added the "secure:<<ENCRYPTED_KEY>>" line to .travis.yml and changed the application.properties in order to set yandex.api_key=VAR_NAME **BUT** when I try to check the code with gradle check, it throws me errors for the tests, caused by "com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException" which I believe is because I don't correctly link the encrypted key with the VAR_NAME variable. 

Has anybody had this problem ?
Thanks

**Comment by fjlopez** [comment]
Why do you se this command? The approach is to tell Travis [to encrypt the file](https://docs.travis-ci.com/user/encrypting-files/).

**Comment by catalindumitrache76** [comment]
Even with file encrypting I keep getting the same error from Travis ... 

The command "openssl aes-256-cbc -K $encrypted_48f484b95ce9_key -iv $encrypted_48f484b95ce9_iv -in application.properties.enc -out src\main\resources\application.properties -d" failed and exited with 1 during .

**Comment by deerter** [comment]
Ok, above the error you mention in Travis there must be some lines explaining what the error actually is (Probably something along the lines of "bad decrypt" or "iv undefined")

**Comment by catalindumitrache76** [comment]
Yeah, you are right ... Travis complains about "bad decrypt":

> bad decrypt
> 139845273269920:error:0606506D:digital envelope routines:EVP_DecryptFinal_ex:wrong final block length:evp_enc.c:532:

**Comment by deerter** [comment]
Ok.
First are you doing this on windows or in Linux?.
If you`re using windows check your ruby version. If it isn´t 2.2 then install it, you can find it in this [link] (http://rubyinstaller.org/downloads/). Once you do that, set the appropiate environment variables in your system and install travis using the command gem install travis. Then do the encryption again and you should be fine (at least that worked for me)

If you're using Linux, then you can check which version of ruby Travis needs in this [link] (https://github.com/travis-ci/travis.rb#other-unix-systems). Can´t be of much help here since i didn´t try it using Linux.

**Comment by catalindumitrache76** [comment]
I had Ruby 2.3 ... i'll try with 2.2 
Thanks for the feedback ;)

**Comment by deerter** [comment]
I also had 2.3 and i had problems installing travis with gems. I managed to install it with that version after many tries but something must have been left out because i couldn´t decrypt either.

**Comment by catalindumitrache76** [comment]
I'm getting the same error over and over ... I tried to modify the .travis.yml file because I figured out that each time you crypt a file with the option --add, 2 lines are added in .travis.yml ... so as we had 2 lines from the beggining, what i did is delete those lines and only let the ones travis generated for me ... sorry for bothering though

**Comment by deerter** [comment]
You have to leave the line that was already there, the only thing you have to edit in that line is the number corresponding to the keys.

**Comment by catalindumitrache76** [comment]
Yeah, that's what I had in the beggining, before going crazy about it xD
I'll talk to Javier next week, thanks for the help though 👍

**Comment by deerter** [comment]
What version of gems do you have?

**Comment by catalindumitrache76** [comment]
I've got version travis version 1.8.2, if that's what you mean ... 
and ruby version 2.2.5

**Comment by deerter** [comment]
type gem -v

**Comment by catalindumitrache76** [comment]
2.4.5.1

**Comment by deerter** [comment]
then you have the same one as me, what do you have in the .yml file?

**Comment by catalindumitrache76** [comment]
This is my .travis.yml file:
`language: java
jdk:
- oraclejdk8
before_install:
- openssl aes-256-cbc -K $encrypted_48f484b95ce9_key -iv $encrypted_48f484b95ce9_iv
  -in src/main/resources/application.properties.enc -out src/main/resources/application.properties
  -d`

**Comment by catalindumitrache76** [comment]
where both _key and _iv are generated by the tool travis encrypt-file application.properties

**Comment by deerter** [comment]
Seems fine to me. Are you on windows 7 or windows 10?

**Comment by catalindumitrache76** [comment]
windows 10 haha

**Comment by deerter** [comment]
Well maybe that´s the problem, I did it on windows 7 so on 10 you may need a different version of the tools. I'm sorry but i can´t be of help anymore. The only thing i can suggest is beginning from scratch on a virtual box with some linux distribution.

**Comment by catalindumitrache76** [comment]
Thank you very much !
